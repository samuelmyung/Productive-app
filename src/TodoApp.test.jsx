import { describe, it, expect } from "vitest";

import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import TodoApp from "./TodoApp";

function addTodo(container, title = 'test title', description = 'test description', priority = 1) {
    const titleInput = container.querySelector('[name=title]');
    const descriptionInput = container.querySelector('[name=description]');
    const priorityInput = container.querySelector('[name=priority]');

    fireEvent.change(titleInput, { target: { value: title }});
    fireEvent.change(descriptionInput, { target: { value: description }});
    fireEvent.change(priorityInput, { target: { value: priority }});

    const button = container.querySelector('.NewTodoForm-addBtn');
    fireEvent.click(button);
}

it('matches snapshot with initial todos', function () {
    const { container } = render(<TodoApp initialTodos={[]} />);
  expect(container).toMatchSnapshot();
});

describe('adding todos', function () {
    it('can add a new todo', function() {
        const { container, queryByText } = render(<TodoApp initialTodos={[]} />);

        expect(queryByText('You have no todos.')).toBeInTheDocument()

        addTodo(container);

        const todo = container.querySelector('.Todo');
        expect(todo).toBeInTheDocument();
        expect(todo).toHaveTextContent('test title');
        expect(todo).toHaveTextContent('test description');
        expect(todo).toHaveTextContent('priority: 1');

    });

    it("matches snapshot after adding box", function () {
        const { container } = render(<TodoApp initialTodos={[]} />);
        addTodo(container);
        expect(container).toMatchSnapshot();
      })
})

describe('removing todos', function () {
    it('can remove a todo', function() {
        const { container } = render(<TodoApp initialTodos={[]} />);
        addTodo(container);

        const removeBtn = container.querySelector('.EditableTodo-delBtn');

        fireEvent.click(removeBtn);
        expect(removeBtn).not.toBeInTheDocument();
    });

    it('matches snapshot after removing todo', function () {
        const { container } = render(<TodoApp initialTodos={[]} />);
        addTodo(container);
        fireEvent.click(container.querySelector('.EditableTodo-delBtn'));
        expect(container).toMatchSnapshot();
    });
});