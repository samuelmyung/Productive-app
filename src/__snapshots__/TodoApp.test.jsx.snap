// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`adding todos > matches snapshot after adding box 1`] = `
<div>
  <main
    class="TodoApp"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
              >
                <div>
                  <b>
                    test title
                  </b>
                   
                  <small>
                    (priority: 
                    1
                    )
                  </small>
                </div>
                <div>
                  <small>
                    test description
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <section
          class="mb-4"
        >
          <h3>
            Top Todo
          </h3>
          <div
            class="Todo"
          >
            <div>
              <b>
                test title
              </b>
               
              <small>
                (priority: 
                1
                )
              </small>
            </div>
            <div>
              <small>
                test description
              </small>
            </div>
          </div>
        </section>
        <section>
          <h3
            class="mb-3"
          >
            Add Nü
          </h3>
          <form
            class="NewTodoForm"
          >
            <div
              class="mb-3"
            >
              <input
                aria-label="Title"
                class="form-control"
                id="newTodo-title"
                name="title"
                placeholder="Title"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <textarea
                aria-label="Description"
                class="form-control"
                id="newTodo-description"
                name="description"
                placeholder="Description"
              />
            </div>
            <div
              class="mb-3 d-flex justify-content-between"
            >
              <div
                class="w-75 d-flex justify-content-between"
              >
                <label
                  class="d-inline-flex"
                  for="newTodo-priority"
                >
                  Priority:  
                </label>
                <select
                  class="form-control form-control-sm d-inline-flex"
                  id="newTodo-priority"
                  name="priority"
                >
                  <option
                    value="1"
                  >
                    Ultra-Über
                  </option>
                  <option
                    value="2"
                  >
                    Über
                  </option>
                  <option
                    value="3"
                  >
                    Meh
                  </option>
                </select>
              </div>
              <button
                class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
              >
                Gø!
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</div>
`;

exports[`matches snapshot with initial todos 1`] = `
<div>
  <main
    class="TodoApp"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <span
          class="text-muted"
        >
          You have no todos.
        </span>
      </div>
      <div
        class="col-md-6"
      >
        <section>
          <h3
            class="mb-3"
          >
            Add Nü
          </h3>
          <form
            class="NewTodoForm"
          >
            <div
              class="mb-3"
            >
              <input
                aria-label="Title"
                class="form-control"
                id="newTodo-title"
                name="title"
                placeholder="Title"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <textarea
                aria-label="Description"
                class="form-control"
                id="newTodo-description"
                name="description"
                placeholder="Description"
              />
            </div>
            <div
              class="mb-3 d-flex justify-content-between"
            >
              <div
                class="w-75 d-flex justify-content-between"
              >
                <label
                  class="d-inline-flex"
                  for="newTodo-priority"
                >
                  Priority:  
                </label>
                <select
                  class="form-control form-control-sm d-inline-flex"
                  id="newTodo-priority"
                  name="priority"
                >
                  <option
                    value="1"
                  >
                    Ultra-Über
                  </option>
                  <option
                    value="2"
                  >
                    Über
                  </option>
                  <option
                    value="3"
                  >
                    Meh
                  </option>
                </select>
              </div>
              <button
                class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
              >
                Gø!
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</div>
`;

exports[`removing todos > matches snapshot after removing todo 1`] = `
<div>
  <main
    class="TodoApp"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <span
          class="text-muted"
        >
          You have no todos.
        </span>
      </div>
      <div
        class="col-md-6"
      >
        <section>
          <h3
            class="mb-3"
          >
            Add Nü
          </h3>
          <form
            class="NewTodoForm"
          >
            <div
              class="mb-3"
            >
              <input
                aria-label="Title"
                class="form-control"
                id="newTodo-title"
                name="title"
                placeholder="Title"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <textarea
                aria-label="Description"
                class="form-control"
                id="newTodo-description"
                name="description"
                placeholder="Description"
              />
            </div>
            <div
              class="mb-3 d-flex justify-content-between"
            >
              <div
                class="w-75 d-flex justify-content-between"
              >
                <label
                  class="d-inline-flex"
                  for="newTodo-priority"
                >
                  Priority:  
                </label>
                <select
                  class="form-control form-control-sm d-inline-flex"
                  id="newTodo-priority"
                  name="priority"
                >
                  <option
                    value="1"
                  >
                    Ultra-Über
                  </option>
                  <option
                    value="2"
                  >
                    Über
                  </option>
                  <option
                    value="3"
                  >
                    Meh
                  </option>
                </select>
              </div>
              <button
                class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
              >
                Gø!
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</div>
`;
